### PR TITLE
refactor(website): Upgrade dependencies and use `StarlightPage` component

### PR DESCRIPTION
**Pull request by Yan-Thomas**
## Summary

This PR uses the new `<StarlightPage>` component instead of the custom `<StarlightSplashLayout>` one, which was making it impossible for us to upgrade Starlight, Astro and its dependencies.

I've added per-file comments explaining a few changes necessary due to the breaking changes from upgrading.

## Test Plan

Visually expecting the website and seeing everything is working as usual

### CODE REVIEW

File: website/astro.config.ts [hunk 1]
```diff
@@ -1,40 +1,9 @@
-import { netlifyStatic } from "@astrojs/netlify";
```
Comment (Yan-Thomas) [inline website/astro.config.ts:1]: The static Netlify adapter will be deprecated, so I've removed it. Note that we technically don't even need this one, so we should feel safe to remove it as well.

File: website/astro.config.ts [hunk 2]
```diff
@@ -1,40 +1,9 @@
...
+import netlify from "@astrojs/netlify";
 import react from "@astrojs/react";
 import starlight from "@astrojs/starlight";
 import { defineConfig } from "astro/config";
-import { h } from "hastscript";
-import { escape as htmlEscape } from "html-escaper";
-import rehypeAutolinkHeadings from "rehype-autolink-headings";
+import { rehypeAutolink } from "./plugins/rehype-autolink";
 import rehypeSlug from "rehype-slug";
-import remarkToc from "remark-toc";
```
Comment (Yan-Thomas) [inline website/astro.config.ts:9]: Looks like we weren't using `"remark-toc"` anywhere, so I've removed it. Also, did the same for `@astrojs/vercel`, which isn't being used anymore.

File: website/astro.config.ts [hunk 3]
```diff
@@ -417,6 +381,7 @@ export default defineConfig({
 
 		server: {
 			fs: {
+				strict: false,
```
Comment (Yan-Thomas) [inline website/astro.config.ts:384]: Vite is having some issue with the DevToolbar trying to import its code and failing because of the allow list configuration, disabling `strict` works, but would appreciate if anyone was able to fix it properly in the `allow` setting

File: website/astro.config.ts [hunk 4]
```diff
@@ -397,11 +362,10 @@ export default defineConfig({
 
 	markdown: {
 		syntaxHighlight: "prism",
-		remarkPlugins: [remarkToc],
-		rehypePlugins: [rehypeSlug, [rehypeAutolinkHeadings, autolinkConfig]],
+		rehypePlugins: [rehypeSlug, ...rehypeAutolink()],
```
Comment (Yan-Thomas) [inline website/astro.config.ts:365]: Since the `unified` project had breaking changes, we now have some weird behavior clashing issue when trying to apply the `remarkAutolink` plugin, had to move it to an external file to make it work properly.

File: website/src/components/InlineSVG.astro [hunk 5]
```diff
@@ -1,18 +1,19 @@
 ---
 export interface Props {
-	src: string;
+  src: string;
 }
 
 const { src } = Astro.props as Props;
 
 const files = await import.meta.glob<string>("/src/assets/svg/**/*.svg", {
-	as: "raw",
```
Comment (Yan-Thomas) [inline website/src/components/InlineSVG.astro:9]: `as: "raw"` was deprecated in Vite, so I've replaced it for the recommended `query: "?raw"`

File: website/src/playground/PlaygroundLoader.tsx [hunk 6]
```diff
@@ -343,7 +343,7 @@ function initState(
 				searchParams.get("bracketSpacing") === "true" ||
 				defaultPlaygroundState.settings.bracketSpacing,
 			bracketSameLine:
-				searchParams.get("bracketSameLine") === "true" ??
+				searchParams.get("bracketSameLine") === "true" ||
```
Comment (Yan-Thomas) [inline website/src/playground/PlaygroundLoader.tsx:346]: Vite was complaining and throwing an warning about this line where the left operand would always be used, might be a false alarm from it, so that's something we can change back, probably.

File: website/astro.config.ts [hunk 7]
```diff
@@ -397,11 +362,10 @@ export default defineConfig({
...
 	},
 
-	adapter: netlifyStatic(),
+	adapter: netlify(),
 
 	vite: {
```
Comment (Yan-Thomas) [inline website/astro.config.ts:370]: For some reason Vite is complaining about some imported code by Astro's DevToolbar. Supposedly `vite.server.fs.allow` the path used should fix it, but didn't work for me.

File: website/package.json [hunk 8]
```diff
@@ -34,25 +33,25 @@
...
 		"postcss": "^8.4.30",
 		"prettier": "^3.1.0",
-		"react": "^17.0.2",
-		"react-dom": "^17.0.2",
-		"rehype-autolink-headings": "^6.1.1",
-		"rehype-slug": "^5.1.0",
-		"rehype-toc": "^3.0.2",
-		"remark-toc": "^8.0.1",
+		"react": "^18.2.0",
+		"react-dom": "^18.2.0",
```
Comment (ematipico) [inline website/package.json:51]: Seems like the playground doesn't work anymore. Unless you want to spend time understanding the issue, maybe we can revert the change

File: website/astro.config.ts (hunk 7 above)
Comment (ematipico) [reply website/astro.config.ts:370]: We have to disable their remote image service, because they don't support multiple remote image domains from the config file. Their alternative is to use another configuration file

File: website/src/playground/PlaygroundLoader.tsx (hunk 6 above)
Comment (ematipico) [reply website/src/playground/PlaygroundLoader.tsx:346]: We should revert this change, it previously caused some issue

File: website/astro.config.ts (hunk 1 above)
Comment (ematipico) [reply website/astro.config.ts:1]: We don't??? 😯🤯

File: website/package.json (hunk 8 above)
Comment (Yan-Thomas) [reply website/package.json:51]: The issue doesn't seem related to updating the React dependencies, reverting them doesn't fix, nor does loading the Playground without `<StarlightPage>` - my best guess it might be Vite-related? But hard to know without context on how the Playground was built.

File: website/astro.config.ts (hunk 1 above)
Comment (Yan-Thomas) [reply website/astro.config.ts:1]: Nope, both Vercel and Netlify let you deploy just fine without an adapter, we do mention it in the docs: https://docs.astro.build/en/guides/integrations-guide/netlify/

It might enable some host-specific features but for the most part, the adapters aren't focused on the static use case.

File: website/package.json [hunk 9]
```diff
@@ -34,25 +33,25 @@
 		"@types/node": "^18.18.0",
 		"@types/react": "^18.2.23",
 		"@types/react-dom": "^18.2.8",
-		"@uiw/react-codemirror": "4.20.2",
+		"@uiw/react-codemirror": "4.21.21",
```
Comment (Sec-ant) [inline website/package.json:36]: The playground not working anymore is caused by this bump.

This version of `@uiw/react-codemirror` indirectly requires the version of `@codemirror/state` in range `^0.6.4` because of its dependencies: `@codemirror/view` and `@codemirror/commands`.

However [`@codemirror/state@6.2.1` is explicitly pinned in the `devDependencies`](https://github.com/biomejs/biome/blob/295f4bd4d46efe6b6927c2557b1b3292bd704806/website/package.json#L28), so two versions of `@codemirror/state` are installed and hence the [error occurs](https://github.com/uiwjs/react-codemirror/issues/216) which you can find in the console, and it breaks the playground.

A simple fix is to bump the `@codemirror/state` in `devDepedencies` to `0.6.4`.

Comment (Yan-Thomas) [reply website/package.json]: Nice catch, thanks @Sec-ant - hadn't noticed the dependencies were pinned, I've moved them to ranges instead and seem to be working fine locally!

File: website/astro.config.ts (hunk 7 above)
Comment (Yan-Thomas) [reply website/astro.config.ts:373]: There we go, this the `vite.server.fs.allow` should be fixed, Vite now has a utility to allow workspace packages to be served, which fixes the issue with serving files from dependencies.

**Review by ematipico — APPROVED** [review]

**Comment by ysm-dev** [comment]
RSS link broken after this PR: https://biomejs.dev/feed.xml
